pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```diff
@@ -0,0 +1,319 @@
+                username, password);
+        reader.read();
+    }
+
+    private static Options getOptions() {
+        final Options options = new Options();
+        options.addOption("s", "scope", true, "The scope name of the stream to read from.");
+        options.addOption("n", "stream", true, "The name of the stream to read from.");
+        options.addOption("u", "uri", true, "The URI to the controller in the form tls://host:port");
+        options.addOption("t", "truststore", true,
+                "The location of .pem truststore file in the file system to use by this application process.");
+        options.addOption("v", "validatehost", false, "Whether to verify server's hostname.");
+        options.addOption("a", "username", true,
+                "The account username to use by the client for authenticating to the server.");
+        options.addOption("p", "password", true,
+                "The account password to use by the client for authenticating to the server.");
+        return options;
+    }
+
+    private static CommandLine parseCommandLineArgs(Options options, String[] args) throws ParseException {
+        CommandLineParser parser = new DefaultParser();
+        CommandLine cmd = parser.parse(options, args);
+        return cmd;
+    }
+
+    public void run() {
+        LOG.info("Starting segment logger");
+        this.executor = newScheduledThreadPool(THREAD_POOL_SIZE, String.format("EventsLogger"));
+        try {
+            Futures.delayedTask(() -> {
+                logEventCounts();
+                this.executor.shutdownNow();
+                return null;
+            }, Duration.ofMinutes(0), executor).get();
+        } catch (InterruptedException | ExecutionException e) {
+           e.printStackTrace();
+        }
+        LOG.info("Started segment logger");
+    }
+
+    private void logEventCounts() {
+        try (BatchClientFactory batchClient = BatchClientFactory.withScope(scope, clientConfig)) {
+             System.out.println("Done creating batchClient for " + scope + "/" + stream);
+             ArrayList<SegmentRange> ranges = Lists.newArrayList(batchClient.getSegments(Stream.of(scope, stream), StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+             readFromRanges(batchClient, ranges);
+             System.out.println("Done reading ranges of size: " + String.valueOf(ranges.size()));
+        } catch (Exception e) {
+            e.printStackTrace();
+            LOG.error("Exception:", e);
+        }
+    }
+
+    private int readFromRanges(BatchClientFactory batchClient, List<SegmentRange> ranges) {
+        LOG.info("Number of ranges: {}", ranges.size());
+        List<CompletableFuture<Integer>> eventCounts = new ArrayList<>();
+        JavaSerializer<String> serializer = new JavaSerializer<String>();
+        eventCounts = ranges
+                .parallelStream()
+                .map(range -> CompletableFuture.supplyAsync(() -> batchClient.readSegment(range, serializer))
```

Comment:
You are executing this on the `ForkJoinPool.commonPool()` executor. Same with the callback. Switch it to use one of your executors.

Reply:
Also, why do you execute the `readSegment` in a `supplyAsync` and then use `thenApply`? None of these are async; you can simplify your code by combining these in the same block of code.

Reply:
Yes Andre